javaparser-core/src/main/java/com/github/javaparser/utils/CollectionStrategy.java

```diff
@@ -48,26 +48,25 @@
             final JavaParser javaParser = new JavaParser(getParserConfiguration());
             final ParseResult<CompilationUnit> parseResult = javaParser.parse(file);
 
+
             if (parseResult.isSuccessful()) {
-                if (parseResult.getProblems().isEmpty()) {
-                    if (parseResult.getResult().isPresent()) {
-                        final CompilationUnit compilationUnit = parseResult.getResult().get();
-                        final Optional<CompilationUnit.Storage> storage = compilationUnit.getStorage();
-                        if (storage.isPresent()) {
-                            final Optional<Path> sourceRootPath = storage.map(CompilationUnit.Storage::getSourceRoot);
-                            return sourceRootPath;
-                        } else {
-                            Log.info("Storage information not present -- an issue with providing a string rather than file reference?");
+                if (parseResult.getResult().isPresent()) {
+                    final Optional<CompilationUnit.Storage> storage = parseResult.getResult().flatMap(CompilationUnit::getStorage);
+                    if (storage.isPresent()) {
+                        if (storage.get().getFileName().equals("module-info.java")) {
```

Comment:
Ran into an older issue with module-info.java confusing the ParserCollectionStrategy. The issue was marked as fixed, but it sure failed badly here :-)